CONTRIBUTING.md

```diff
@@ -0,0 +1,232 @@
+```bash
+git clone https://github.com/ansible/ansible.git
+```
+
+Instead of installing ansible-core from source, you can also work with an already existing installation of Ansible, ansible-base or ansible-core. Simply skip steps 2 and 3 in that case.
+
+3. Go to the cloned repository and prepare the environment:
+```bash
+cd ansible && source hacking/env-setup
+pip install -r requirements.txt
+cd ~
+```
+4. Create the following directories in your home directory:
+```bash
+mkdir -p ~/ansible_collections/NAMESPACE/COLLECTION_NAME
+```
+
+For example, if the collection is `community.mysql`, it will be:
+```bash
+mkdir -p ~/ansible_collections/community/mysql
+```
+
+If the collection is `ansible.posix`, it will be:
+```bash
+mkdir -p ~/ansible_collections/ansible/posix
+```
+
+5. Fork the collection repository through the GitHub web interface.
+
+6. Clone the forked repository from your profile to the created path:
+```bash
+git clone https://github.com/YOURACC/COLLECTION_REPO.git ~/ansible_collections/NAMESPACE/COLLECTION_NAME
+```
+
+If you prefer to use the SSH protocol:
+```bash
+git clone git@github.com:YOURACC/COLLECTION_REPO.git ~/ansible_collections/NAMESPACE/COLLECTION_NAME
+```
+
+
+7. Go to your new cloned repository.
+```bash
+cd ~/ansible_collections/NAMESPACE/COLLECTION_NAME
+```
+
+8. Be sure you are in the default branch (it is usually `main`):
+```bash
+git status
+```
+
+9. Show remotes. There should be the `origin` repository only:
+```bash
+git remote -v
+```
+
+10. Add the `upstream` repository:
+```bash
+git remote add upstream https://github.com/ansible-collections/COLLECTION_REPO.git
+```
+
+11. Update your local default branch. Assuming that it is `main`:
+```bash
+git fetch upstream
+git rebase upstream/main
+```
+
+12. Create a branch for your changes:
+```bash
+git checkout -b name_of_my_branch
+```
+
+13. We recommend you start with writing integration tests if applicable.
+
+Note: If there are any difficulties with writing / running the tests or you are not sure if the case can be covered, feel free to skip this step.
+If needed, other contributors can help you with it later.
+
+Note: Some collections do not have integration tests.
+
+All integration tests are stored in `tests/integration/targets` subdirectories.
+Go to the subdirectory containing the name of module you are going to change.
+For example, if you are fixing the `mysql_user` module in the `community.mysql` collection,
+its tests are in `tests/integration/targets/test_mysql_user/tasks`.
+
+The `main.yml` file holds test tasks and includes other test files.
+Look for a suitable test file to integrate your tests or create and include a dedicated test file.
+You can use one of the existing test files as a draft.
+
+When fixing a bug, write a task which reproduces the bug from the issue.
+
+Put the reported case in the tests, then run integration tests with the following command:
+```bash
+ansible-test integration name_of_test_subdirectory --docker -vvv
```

Comment:
Usually `-v` makes the output a lot more readable, especially if there already are extensive tests.

Reply:
as far as i remember, when only one `v`, there are no return values, so a bit difficult to compare what is in an assertion with what a module returned

Reply:
A single `-v` prints return values, though not nicely formatted (i.e. no JSON pretty-printing). I usually run it with `-v`, and if that fails and it's too annoying to read, I re-run with `-vvv` :)

Reply:
Ok, changed:)
I also added
```
You can use the `-vv` or `-vvv` argument, if you need more detailed output.
```